Sources/XcodeGenKit/SettingsBuilder.swift

```diff
@@ -24,18 +24,47 @@ extension ProjectSpec {
         }
 
         buildSettings += getBuildSettings(settings: settings, config: config)
```

Comment:
I think the clearing should happen before this line, so custom build settings don't get removed, just the setting presets